index.js

```diff
@@ -4030,6 +4037,29 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# sortBy :: (a -> a -> Ordering) -> Array a -> Array a
+  //.
+  //. Returns a copy of the array, sorted according to the comparator
+  //. function. The comparator function is expected to be curried,
+  //. and will receive two values at a time. The comparator must return:
+  //. - 'LT': if the first value is **less** than the second value.
+  //. - 'GT': if the first value is **greater** than the second value.
+  //. - 'EQ': if the first value is **equal** to the second value.
+  //.
+  //. ```javascript
+  //. > function increasingOrder(a) { return function(b) { return a - b; }; }
+  //.
+  //. > S.sortBy(increasingOrder)([4, 2, 7, 5]);
+  //. [2, 4, 5, 7]
+  //. ```
+  function sortBy(cmp, xs) {
+    return xs.slice().sort(function(a, b) {
+      return cmp(a)(b);
```

Comment:
Any idea on how to make the comparison function optionally curried? as in `cmp(a, b)` and `cmp(a)(b)`?